datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx

```diff
@@ -0,0 +1,327 @@
+        );
+    };
+
+    const tagSearchOptions = tagSearchResults.map((result) => {
+        return renderSearchResult(result);
+    });
+
+    const inputExistsInTagSearch = tagSearchResults.some((result: SearchResult) => {
+        const displayName = entityRegistry.getDisplayName(result.entity.type, result.entity);
+        return displayName.toLowerCase() === inputValue.toLowerCase();
+    });
+
+    if (!inputExistsInTagSearch && inputValue.length > 0 && type === EntityType.Tag && urnIds.length === 0) {
+        tagSearchOptions.push(
+            <Select.Option value={CREATE_TAG_VALUE} key={CREATE_TAG_VALUE}>
+                <Typography.Link> Create {inputValue}</Typography.Link>
+            </Select.Option>,
+        );
+    }
+
+    const tagRender = (props) => {
+        // eslint-disable-next-line react/prop-types
+        const { label, value, closable, onClose } = props;
+        const { type: selectedType } = getSelectedValue(value);
+        const onPreventMouseDown = (event) => {
+            event.preventDefault();
+            event.stopPropagation();
+        };
+        return (
+            <CustomTag
+                onMouseDown={onPreventMouseDown}
+                closable={closable}
+                onClose={onClose}
+                style={{
+                    padding: selectedType === EntityType.Tag ? '0px 7px 0px 0px' : '0px 7px',
+                    marginRight: 3,
+                    borderRadius: '100em',
+                    lineHeight: 0,
+                    display: 'flex',
+                    justifyContent: 'start',
+                    alignItems: 'center',
+                }}
+            >
+                {label}
+            </CustomTag>
+        );
+    };
+
+    // Handle the Enter press
+    useEnterKeyListener({
+        querySelectorToExecuteClick: '#addTagButton',
+    });
+
+    if (showCreateModal) {
+        return (
+            <CreateTagModal
+                visible={visible}
+                onClose={onCloseModal}
+                onBack={() => setShowCreateModal(false)}
+                tagName={inputValue}
+                entityUrn={entityUrn}
+                entitySubresource={entitySubresource}
+            />
+        );
+    }
+
+    const getUrnId = (urnId: string) => {
+        const { name: selectedName, type: selectedType } = getSelectedValue(urnId);
+        let tempUrn = '';
+        if (selectedType === EntityType.Tag) {
+            tempUrn = `urn:li:tag:${selectedName}`;
+        }
+        if (selectedType === EntityType.GlossaryTerm) {
+            tempUrn = `urn:li:glossaryTerm:${selectedName}`;
+        }
+        return tempUrn;
+    };
+
+    // When a Tag or term search result is selected, add the urn to the UrnIds
+    const onSelectValue = (newUrnId: string) => {
+        if (newUrnId === CREATE_TAG_VALUE) {
+            setShowCreateModal(true);
+            return;
+        }
+        const newUrnIds = [...(urnIds || []), getUrnId(newUrnId)];
```

Comment:
Let's rename UrnIds to just Urns, if this array just holds URNs.

Reply:
Changed and committed

Reply:
Changed and Committed my changes